amethyst_assets/src/storage.rs

```diff
@@ -153,7 +153,7 @@ impl<A: Asset> AssetStorage<A> {
                     }).and_then(|(d, rel)| f(d).map(|a| (a, rel)))
                         .chain_err(|| ErrorKind::Asset(name.clone()))
                     {
-                        Ok(x) => {
+                        | Ok(x) => {
```

Comment:
why?

Reply:
Here's why I'm asking you to revert this change:

* not supported by older Rust versions
* this is not even the recommended style
* it doesn't look very nice neither is it consistent with the other code